### DISCUSSION TITLE
Notes from TF Meeting 20.02.2023: PVD schema and workflow schema discussion

### DISCUSSION THREAD

**Discussion by budschi** [FAIRmat-NFDI/AreaA-data_modeling_and_schemas#34] [Task Force Meeting]
- Github discussion for TF notes  
- PVD  schema presentation by @hampusnasstrom 
	- find it here:  https://github.com/FAIRmat-NFDI/AreaA-data_modeling_and_schemas/tree/add_pvd/PVD 
	- structure of schema as in Readme.md  
	- example data which runs in NOMAD staging  
	- explaining schema in NOMAD with inheritance, overwritten  
	- sources  
	- time series --> issue: https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/issues/1352#note_172920  
	- Thermal PVD  
		- Difference between thermal source and pvd source  
	- example data in NOMAD  
	
	- Discussions on: 
	  - Recipe? or Setppoints? Steps?
	  - generic source class meeting for different techniques in the Area  
	  - sources context  
	     - Activity --> process --> synthesis --> vapor deposition --> pvd  
	- visualization of schema tool nedded? sofar done manually by Andrea --> https://github.com/FAIRmat-NFDI/AreaA-data_modeling_and_schemas/tree/main/base_classes   
- Discussion on workflow:  
	- example of powder preperation --> graph of Hampus  
	- https://nomad-lab.eu/dev/rae/dev-1346-add-feat-fcvmwq/gui/  
		- create  my workflow  
	- *data* and *workflow* section  
		- workflow has only inputs, outputs, tasks  
		- 1. option: mimick worklfow structure in synthesis schema? (like Andreas examples) and then copy it over  
		   2. option: normalizer that transfer *user-friendly* structure into workflow section

	- do we something like a *sample* section  
			- problem of substrate section vs. genereic solution with components  
	- make use of *experimental workflow* section  
	- need for *experimental planer*

**Comment by budschi** [comment]
@hampusnasstrom could you add your example picture for the workflow of powder preparation you showed? and also your quick example of option1 and option2 of the workflow and normalizer implementation (if you saved it)? Thanks

**Comment by hampusnasstrom** [comment]
Here is the example workflow for the powder preparation:
![image](https://user-images.githubusercontent.com/16759628/220282417-813fe4a3-2541-483c-877f-8c2dc7ff69dd.png)

Here is option 1:
```json
{
    "data": {
        "inputs": [
            "CsBr",
            "CsI",
            "Substrate"
        ],
        "outputs": [
            "sample"
        ],
        "tasks": [
            "Source_heat_up",
            "deposition",
            "cooldown"
        ]
    },
    "workflow2": {
        "inputs": [
            "CsBr",
            "CsI",
            "Substrate"
        ],
        "outputs": [
            "sample"
        ],
        "tasks": [
            "Source_heat_up",
            "deposition",
            "cooldown"
        ]
    }
}
```

and here is option 2:
```json
{
    "data": {
        "source_materials": [
            "CsBr",
            "CsI"
        ],
        "substrate": [
            "Substrate"
        ],
        "sample": [
            "sample"
        ],
        "procedure_steps": [
            "Source_heat_up",
            "deposition",
            "cooldown"
        ]
    },
    "workflow2": {
        "inputs": [
            "CsBr",
            "CsI",
            "Substrate"
        ],
        "outputs": [
            "sample"
        ],
        "tasks": [
            "Source_heat_up",
            "deposition",
            "cooldown"
        ]
    }
}
```
